src/studio/src/designer/backend/Controllers/RepositoryController.cs

```diff
@@ -401,7 +401,12 @@ public async Task<ActionResult<RepositoryModel>> CopyApp(string org, string sour
                     return Created(repo.CloneUrl, repo);
                 }
 
-                await _repository.DeleteRepository(org, targetRepository);
+                // Avoid deleting pre-existing repositories that have resulted in a naming conflict.
+                if (repo.RepositoryCreatedStatus != HttpStatusCode.Conflict)
+                {
+                    await _repository.DeleteRepository(org, targetRepository);
```

Comment:
Could we still end up deleting an existing repo here? What if the request for some reason times out or some glitch causes a 500 response even if the repo exists.

I guess the chances of something like that happening are slim to none, but maybe a cleaner way would be an explicit check for repo name existing? Thoughts?

Reply:
Or am I simply overthinking this?

Reply:
probably an edge case, but then again our user's might make a mistake.. We can add a service for checking if a repo exists before attempting to copy I guess.. and then let all other cases fall through to deletion on error.